hypershift-operator/controllers/nodepool/kubevirt/kubevirt.go

```diff
@@ -246,3 +233,124 @@ func MachineTemplateSpec(image string, nodePool *hyperv1.NodePool) *capikubevirt
 		},
 	}
 }
+
+// todo: move up when finalized (?)
+const (
+	cdiImmediateBindingAnnotation = "cdi.kubevirt.io/storage.bind.immediate.requested"
+	// todo: use real values
+	bootImagePVCLabelApp       = "app"         // temp value
+	bootImagePVCLabelAppValue  = "hypershift"  // temp value
+	bootImagePVCLabelType      = "type"        // temp value
+	bootImagePVCLabelTypeValue = "kubevirt"    // temp value
+	bootImagePVCLabelCluster   = "clusterName" // temp value
+
+	bootImagePVCAnnotationImage = "image" // temp value
+	bootImagePVCPrefix          = "kubevirt-boot-image-"
+)
+
+func CacheImage(ctx context.Context, cl client.Client, bootImage string, nodePool *hyperv1.NodePool) error {
+	kvPlatform := nodePool.Spec.Platform.Kubevirt
+	clusterName := nodePool.Spec.ClusterName
+	namespace := nodePool.GetNamespace()
+	dvName := getCachedPVCName(clusterName)
+
+	pvc := corev1.PersistentVolumeClaim{}
+
+	lbls := map[string]string{
+		bootImagePVCLabelApp:     bootImagePVCLabelAppValue,
+		bootImagePVCLabelType:    bootImagePVCLabelTypeValue,
+		bootImagePVCLabelCluster: clusterName,
+	}
```

Comment:
We need either a label or annotation which stores the bootimage hash. This hash is how we'll determine whether or not a new cached DV needs to be created, or whether we can use a previously cached image.

This [1] is where that bootImage is detected today. It's not immediately obvious, but there's a hash in there as well we can retrieve. we return `disk.Location` url today, but there's also a `disk.sha256` that we have access too. example struct [2].

That sha256 is our key when caching, not the location url. 


1. https://github.com/openshift/hypershift/blob/afcbd08154e05c2f711756e18b7c81e360390f0c/hypershift-operator/controllers/nodepool/kubevirt/kubevirt.go#L26
2. https://github.com/openshift/hypershift/blob/afcbd08154e05c2f711756e18b7c81e360390f0c/support/releaseinfo/releaseinfo.go#L51